### PR TITLE
Don't always set the trail.Failed to true due to how SetValid works

### PR DESCRIPTION
**Pull request by MStoykov**

### CODE REVIEW

**Comment by MStoykov** [comment]
A cautionary tale on RTFM ... https://pkg.go.dev/gopkg.in/guregu/null.v3#Bool.SetValid

**Review by na-- — DISMISSED** [review]
LGTM, though if possible, can you check this in some test?

**Review by imiric — APPROVED** [review]

**Review by na-- — APPROVED** [review]
LGTM, it might be simpler to also add the v0.31.1 release notes in this same PR

File: lib/netext/httpext/request_test.go [hunk 1]
```diff
@@ -280,3 +284,69 @@ func BenchmarkWrapDecompressionError(b *testing.B) {
...
+		"null responsecallback": {responseCallback: nil, failed: null.NewBool(false, false)},
+		"unexpected response":   {responseCallback: func(int) bool { return false }, failed: null.NewBool(true, true)},
+		"expected response":     {responseCallback: func(int) bool { return true }, failed: null.NewBool(false, true)},
+	}
+	for name, testCase := range testCases {
+		responseCallback := testCase.responseCallback
+		failed := testCase.failed
+
+		t.Run(name, func(t *testing.T) {
+			ctx = lib.WithState(ctx, state)
```
Comment (imiric) [inline lib/netext/httpext/request_test.go:322]: Do you need to reattach the state here? It's already done on line 307.

**Review by imiric — DISMISSED** [review]
@na-- I'd leave the release notes for a separate PR.

File: lib/netext/httpext/request_test.go (hunk 1 above)
Comment (MStoykov) [reply lib/netext/httpext/request_test.go:322]: no, I probably just copy-pasted more when I was moving to a testcases variant

**Review by na-- — APPROVED** [review]

**Review by imiric — APPROVED** [review]

**Pull request merged**